Authentication/SignUp/View/SignupController.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  SignupController.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 3/7/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+import ReactiveCocoa
+
+public final class SignupController: UIViewController {
+    
+    private let _viewModel: SignupViewModelType
+    private let _signupViewFactory: () -> SignupViewType
+    private let _delegate: SignupControllerDelegate
+    
+    public lazy var signupView: SignupViewType = self._signupViewFactory()
+    
+    private lazy var _notificationCenter: NSNotificationCenter = NSNotificationCenter.defaultCenter()
+    private var _disposable = CompositeDisposable()
+    private let _keyboardDisplayed = MutableProperty(false)
+    private let _activeTextField = MutableProperty<UITextField?>(.None)
+
+    
+    init(viewModel: SignupViewModelType, signupViewFactory: () -> SignupViewType, delegate: SignupControllerDelegate = DefaultSignupControllerDelegate()) {
+        _viewModel = viewModel
+        _signupViewFactory = signupViewFactory
+        _delegate = delegate
+        super.init(nibName: nil, bundle: nil)
+        addKeyboardObservers()
+    }
+
+    deinit {
+        _disposable.dispose()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public override func loadView() {
+        self.view = signupView.view
+    }
+    
+    
+    public override func viewDidLoad() {
+        signupView.render()
+        bindViewModel()
+        let tapRecognizer: UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+        self.view.addGestureRecognizer(tapRecognizer)
+    }
+    
+}
+
+private extension SignupController {
+    
+    func bindViewModel() {
```

Comment:
we should decide wether we name private or not functions inside a private extension. I think @danielaRiesgo and I agree to use it, wether @guidomb you don't like that. 
While doing code review in greenco, I am more convinced that the function should be private rather than without label. I think we should leave the least possible as ambigouos. However, it's your call @guidomb